internal/gatewayapi/backendtrafficpolicy.go

```diff
@@ -68,10 +68,38 @@ func (t *Translator) ProcessBackendTrafficPolicies(resources *resource.Resources
 
 	handledPolicies := make(map[types.NamespacedName]*egv1a1.BackendTrafficPolicy)
 
+	gatewayPolicyMap := make(map[types.NamespacedName]*egv1a1.BackendTrafficPolicy)
+
 	// Translate
 	// 1. First translate Policies targeting xRoutes
 	// 2. Finally, the policies targeting Gateways
 
+	// Build gateway policy map
+	for _, currPolicy := range backendTrafficPolicies {
+		targetRefs := getPolicyTargetRefs(currPolicy.Spec.PolicyTargetReferences, gateways)
+		for _, currTarget := range targetRefs {
+			if currTarget.Kind == resource.KindGateway {
+				// Check if the gateway exists
+				key := types.NamespacedName{
+					Name:      string(currTarget.Name),
+					Namespace: currPolicy.Namespace,
+				}
+				gateway, ok := gatewayMap[key]
+				if !ok {
+					continue
+				}
+
+				// Check if policy for gateway exists
+				gatewayNN := utils.NamespacedName(gateway)
```

Comment:
this is a tricky line, this logic here is a pre processing step where we are building some state
however, we haven't yet computed which policies targeting a gateway have been accepted,
is this logic of `if exists, skip` safe here ? thinking out loud - it may be since we are sorting based on timestamp on L45

Reply:
the first loop is very similar to the third one(the gateway one), but it's hard to merge them into one AFAICT.

Reply:
Yeah I understand, if you can add a comment here around why it's correct/okay, should help the next person looking at this code

Reply:
we may need to review/refactor the whole thing in `v1.5`.